react/features/overlay/components/OverlayContainer.js

```diff
@@ -117,25 +162,28 @@ class OverlayContainer extends Component {
      * @public
      */
     render() {
-        if (this.props._connectionEstablished && this.props._haveToReload) {
-            return (
-                <PageReloadOverlay
-                    isNetworkFailure = { this.props._isNetworkFailure }
-                    reason = { this.props._reason } />
-            );
-        }
+        const componentMapKey
+            = this.state.filmStripOnly ? 'filmStripOnly' : 'nonFilmStripOnly';
+        let overlayComponent, props;
 
-        if (this.props._suspendDetected) {
-            return (
-                <SuspendedOverlay />
-            );
+        if (this.props._connectionEstablished && this.props._haveToReload) {
+            overlayComponent
+                = filmStripOnlyModeToComponent.pageReload[componentMapKey];
+            props = {
+                isNetworkFailure: this.props._isNetworkFailure,
+                reason: this.props._reason
+            };
+        } else if (this.props._suspendDetected) {
+            overlayComponent
+                = filmStripOnlyModeToComponent.suspended[componentMapKey];
+        } else if (this.props._isMediaPermissionPromptVisible) {
+            overlayComponent = filmStripOnlyModeToComponent
+                .userMediaPermissions[componentMapKey];
+            props = { browser: this.props._browser };
         }
 
-        if (this.props._isMediaPermissionPromptVisible) {
-            return (
-                <UserMediaPermissionsOverlay
-                    browser = { this.props._browser } />
-            );
+        if (overlayComponent !== undefined) {
```

Comment:
do we have rules on how we like to do this? Here you could just check `if (!overlayComponent)`...